catches

ChiefChippy2 opened the pull request:
am tired , but catches mistakes and awaits the promise all and stuff. might wanna test

Commit:
catches

TiltMeSenpai commented on util/command_manager.js:
```diff
@@ -1,34 +1,46 @@
+            case "push":
+                await resolveAll((await CommandList.parse("commands.js")).map(async (command) => {
+                    const cmd_status = await CommandList.put(process.env.DISCORD_APPID, command)
+                    if(cmd_status.ok){
```
This is coming from the Fetch response. If you `await resp.json()`, `cmd_status.ok` doesn't exist here

TiltMeSenpai commented on util/command_manager.js:
```diff
@@ -1,34 +1,46 @@
 const CommandList = require("./CommandList");
+function resolveAll(arr){
+    return Promise.all(arr.map(x=>new Promise(r=>x.catch(_=>console.log("Failed doing Something.")).then(r))))
```
Shouldn't you have access to the error somewhere in here? Pretty sure there's something better than "Failed doing Something." we can return.

ChiefChippy2 replied:
Yes, but it is most probably a fetch error which is multiple lines. Ideally it'd say "failed pushing/deleting (command)" but that needs some refactoring, so I put that as a placeholder

ChiefChippy2 replied:
> Merging for now, we should probably separate command syncing utils from bot utils at some point

Yeah

ChiefChippy2 committed:
Update CommandList.js

Commit:
extended the scripts

ChiefChippy2 commented:
i just added some stuff for listing commands because it wasn't very nice before.
support for guild commands and choices. TODO (sub)options.
Also fixed https://github.com/TiltMeSenpai/cloudflare-bot/pull/1#discussion_r549888147

TiltMeSenpai committed:
Update types to match expected JSDoc strings

TiltMeSenpai commented:
Merging for now, we should probably separate command syncing utils from bot utils at some point
